lib/sidekiq/worker.rb

```diff
@@ -135,8 +140,68 @@ def get_sidekiq_options # :nodoc:
 
       def client_push(item) # :nodoc:
         pool = Thread.current[:sidekiq_via_pool] || get_sidekiq_options['pool'.freeze] || Sidekiq.redis_pool
-        hash = item.stringify_keys
-        Sidekiq::Client.new(pool).push(hash)
+        # stringify
+        item.keys.each do |key|
+          item[key.to_s] = item.delete(key)
+        end
+
+        Sidekiq::Client.new(pool).push(item)
+      end
+
+      def class_attribute(*attrs)
+        instance_reader = true
```

Comment:
Maybe that API could be simplified by using something like `set_property 'prop', value` and `get_property 'prop'` but at least I do understand now why it exists :)